vault/mount.go

```diff
@@ -483,12 +486,19 @@ func (c *Core) decodeMountTable(ctx context.Context, raw []byte) (*MountTable, e
 			continue
 		}
 
+		// Determine type from mount entry
+		pluginType := c.builtinTypeFromMountEntry(ctx, entry)
+
 		// Immediately shutdown the core if deprecated mounts are detected and VAULT_ALLOW_PENDING_REMOVAL_MOUNTS is unset
-		if _, err := c.handleDeprecatedMountEntry(ctx, entry, consts.PluginTypeUnknown); err != nil {
+		if _, err := c.handleDeprecatedMountEntry(ctx, entry, pluginType); err != nil {
 			c.logger.Error("shutting down core", "error", err)
 			c.Shutdown()
 		}
 
+		// Always update the mount entry's Deprecation Status when mounting
+		status, _ := c.builtinRegistry.DeprecationStatus(entry.Type, pluginType)
```

Comment:
In `handleDeprecatedMountEntry` below, `entry.Type` is checked for in `mountAliases` before calling `DeprecationStatus`. Should that `mountAliases` check be done inside `DeprecationStatus` instead so it's harder to write that bug?